tests/integration/ldap_helpers.py

```diff
@@ -0,0 +1,165 @@
+    """Apply an LDIF on glauth-utils."""
+    source_path = f"./tests/integration/data/{ldif_file}"
+    target_path = f"/var/tmp/{ldif_file}"
+    with ops_test.model_context("secondary"):
+        app = kubernetes_model.applications[LDAP_UTILS_APP_NAME]
+        scp_cmd = f"scp {source_path} {app.units[0].name}:{target_path}".split()
+        await ops_test.juju(*scp_cmd)
+        await run_action(kubernetes_model, LDAP_UTILS_APP_NAME, "apply-ldif", path=target_path)
+
+
+async def deploy_glauth(ops_test: OpsTest, kubernetes_model: Model):
+    with ops_test.model_context("secondary"):
+        await asyncio.gather(
+            kubernetes_model.deploy(
+                LDAP_APP_NAME,
+                channel="latest/edge",
+                trust=True,
+                config={"ldaps_enabled": True},
+            ),
+            kubernetes_model.deploy(LDAP_UTILS_APP_NAME, channel="latest/edge", trust=True),
+            kubernetes_model.deploy(
+                POSTGRESQL_K8S,
+                channel="14/stable",
+                trust=True,
+                series="jammy",
+                config={"profile": "testing"},
+                storage={"pgdata": "100G"},
+            ),
+            kubernetes_model.deploy(CERTIFICATES, channel="latest/stable", trust=True),
+            kubernetes_model.deploy(TRAEFIK_CHARM, trust=True),
+        )
+        await kubernetes_model.wait_for_idle(
+            apps=[LDAP_APP_NAME, POSTGRESQL_K8S, CERTIFICATES, LDAP_UTILS_APP_NAME],
+            raise_on_blocked=False,
+        )
+
+        logger.info("Running integrations")
+        await kubernetes_model.integrate(LDAP_APP_NAME, POSTGRESQL_K8S)
+        await kubernetes_model.integrate(LDAP_APP_NAME, CERTIFICATES)
+        await kubernetes_model.integrate(LDAP_APP_NAME, LDAP_UTILS_APP_NAME)
+
+        await kubernetes_model.wait_for_idle(
+            apps=[LDAP_APP_NAME, POSTGRESQL_K8S, CERTIFICATES, LDAP_UTILS_APP_NAME],
+            raise_on_blocked=False,
+        )
+
+        await kubernetes_model.integrate(
+            f"{LDAP_APP_NAME}:ldaps-ingress", f"{TRAEFIK_CHARM}:ingress-per-unit"
+        )
+        await kubernetes_model.wait_for_idle(
+            apps=[LDAP_APP_NAME, CERTIFICATES, TRAEFIK_CHARM],
+            raise_on_blocked=False,  # postgresql can be in Blocked State because of low free space
+            status="active",
+        )
+
+        # On k8s creating an offer on a remote model fails somehow, so we fallback to juju command.
+        first_offer_command = (
+            f"offer {kubernetes_model.info.name}.{LDAP_APP_NAME}:ldap {LDAP_OFFER}"
+        )
+        await ops_test.juju(*first_offer_command.split())
+
+        second_offer_command = (
+            f"offer {kubernetes_model.info.name}.{LDAP_APP_NAME}:send-ca-cert {LDAP_CERT_OFFER}"
+        )
+
+        await ops_test.juju(*second_offer_command.split())
+
+
+async def consume_offers(ops_test: OpsTest, kubernetes_model: Model):
```

Comment:
```suggestion
async def consume_glauth_offers(ops_test: OpsTest, kubernetes_model: Model):
```

Given those are helper classes it would make sense to make it clear to avoid collisions